webui/react/src/pages/ModelDetails.tsx

```diff
@@ -68,6 +70,10 @@ const ModelDetails: React.FC = () => {
     Loaded: (usersPagination) => usersPagination.users,
     NotLoaded: () => [],
   });
+  const ensureWorkspacesFetched = useEnsureWorkspacesFetched(canceler.current);
+  const workspaces = Loadable.getOrElse([], useWorkspaces());
+  const workspace = workspaces.find((ws) => ws.id === model?.model.workspaceId);
```

Comment:
How many workspaces are returned by `useEnsureWorkspacesFetched` - is it possible that there could be a pagination issue and it would be better to have the store retrieve the workspace by ID here?

Reply:
So far, all workspaces. 
[GetWorkspaces](http://latest-master.determined.ai:8080/docs/rest-api/#/Workspaces/GetWorkspaces) doesnt have `ids` params. 
I added `useEnsureWorkspacesFetched` here to follow the patterns somewhere else. Especially when ws name is changes, `useEnsureWorkspacesFetched` can update the data. but i agree that its not efficient. 
The solutions i can think of are 
- Remove `useEnsureWorkspacesFetched` in model and model version cuz even if ws name is updated, it wouldnt be a big issue not to reflect the change.
- Add `workspace ids` in [GetWorkspaces](http://latest-master.determined.ai:8080/docs/rest-api/#/Workspaces/GetWorkspaces)

Reply:
You could add `ensureWorkspaceFetched(id: number)` and `getWorkspace(id: number): O<L<W>>` if you wanted to be defensive.

Reply:
i think it's fine to fetch all ws for now. As Alex said before, the number of ws wouldnt be a lot, and i feel like adding `ensureWorkspaceFetched` in ws store is unnecessary cuz backend can handle it. we have [a filter](https://github.com/determined-ai/determined/blob/master/proto/src/determined/common/v1/common.proto#L34) for this purpose. Currently, the filter is used in [GetExperiments](http://latest-master.determined.ai:8080/docs/rest-api/#/Experiments/GetExperiments), but we can extend it in [GetWorkspaces](http://latest-master.determined.ai:8080/docs/rest-api/#/Workspaces/GetWorkspaces).
@mapmeld what do you think?